### ISSUE TITLE
[flox/update]: Add command `flox update` which updates pinned inputs in existing lockfile.

### ISSUE THREAD

**Issue by aakropotkin** [flox/flox#502]
This allows users to explicitly update pinned inputs in their lockfiles with a command `flox update [input-name]`.

This requires changes to both `pkgdb` and `flox` CLI.

## Behaviors
- When no input is named, update all inputs.
- Command should emit summary of modified pins that is suitable as a commit message.
- Updating does NOT implicitly upgrade packages.

## Future Work
- Updating implicitly kicks off the creation of `pkgdb`s as a background task.
  + Blocked by fixes to SQlite3 database locking/sync.

**Comment by ghudgins** [comment]
implementer's choice on CLI design for this

**Comment by mkenigs** [comment]
@aakropotkin this says it requires changes on the pkgdb side - what's the status of `pkgdb manifest update` and what still needs to be done?

**Comment by mkenigs** [comment]
I don't see any mention of updating the global registry. Is that part of this ticket, or should `update` behave as an `environment` command? If updating the global registry is part of this ticket, how is that specified by the user?

**Comment by mkenigs** [comment]
I would expect this to accept a list of inputs but as stated it only accepts one

**Comment by mkenigs** [comment]
@ghudgins @jennymahmoudi ready for review

**Comment by ghudgins** [comment]
nit: capital `u` in `update an environment's inputs`. just grab it on the next PR?
question: why would we have a remote option for this command? 

besides those, it looked good. tested updating something. seeing the 'All inputs are up to date'. the locking of the global env and updating the global env.

**Comment by mkenigs** [comment]
I'll fix the `u` in an incoming `upgrade` PR.

My understanding is that:
```
# have a remote env at nixpkgs rev 1
flox install -r foo/bar hello
```

Would be different than:
```
# have a remote env at nixpkgs rev 1
flox update -r foo/bar # updates nixpkgs to rev 2
flox install -r foo/bar hello
```

**Comment by ghudgins** [comment]
thanks. okay sounds good!
